components/esp_modem/examples/linux_modem/CMakeLists.txt

```diff
@@ -2,7 +2,8 @@ cmake_minimum_required(VERSION 3.5)
 
 include($ENV{IDF_PATH}/tools/cmake/project.cmake)
 
-set(EXTRA_COMPONENT_DIRS ../../port/linux)
+set(EXTRA_COMPONENT_DIRS ../../port/linux
+            "$ENV{IDF_PATH}/tools/mocks/freertos/")
```

Comment:
Why is this needed? esp_modem should be freertos agnostic, it has an OS abstraction layer, implemented in  `port/linux` for linux.

Reply:
From the log I see it uses freertos
https://github.com/espressif/esp-protocols/actions/runs/7782922741/job/21220340525#:~:text=%2D%2D%20Components%3A%20esp_common%20esp_event_mock,idf/components/soc

Reply:
@david-cermak After a closer look, I found that the `FreeRTOS` is coming as a common component, 
https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/build-system.html#common-component-requirements:~:text=is%3A%20cxx%2C%20newlib%2C-,freertos,-%2C%20esp_hw_support%2C%20heap%2C%20log

https://gitlab.espressif.cn:6688/espressif/esp-idf/-/merge_requests/27455/diffs#338032a49374fdebda46efbb8226c68650fbeee5:~:text=%23%20target.-,%23%20This%20test%20app%20doesn%27t%20require%20FreeRTOS%2C%20using%20mock%20instead,list(APPEND%20EXTRA_COMPONENT_DIRS%20%22%24ENV%7BIDF_PATH%7D/tools/mocks/freertos/%22),-project(nvs_host_test

And as the test doesn't requires freertos, it uses mock instead
https://gitlab.espressif.cn:6688/espressif/esp-idf/-/merge_requests/27455/diffs#338032a49374fdebda46efbb8226c68650fbeee5

Reply:
> And as the test doesn't requires freertos, it uses mock instead 

If it was for the test, I would be okay(-ish) with it, I think adding a comment that the test needs and uses pthreads, but common_components need the FreeRTOS would be probably fine.

But this is an example, use of the freertos mock is simply wrong here.